### PR TITLE
refactor: add `selection.state.draggingArea`

### PR DESCRIPTION
**Pull request by fundon**
part of #1480

### CODE REVIEW

**Comment by fundon** [comment]
It would be better to unify `Point` `Rect`.

e.g. `IPoint`, `IBound`, too many types.

File: packages/blocks/src/__internal__/utils/rect.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { clamp } from '../index.js';
+
 export class Point {
```
Comment (himself65) [inline packages/blocks/src/__internal__/utils/rect.ts:3]: I think we need unit test for the basic class

Comment (himself65) [reply packages/blocks/src/__internal__/utils/rect.ts:3]: please use vitest with happy-dom

Comment (doodlewind) [reply packages/blocks/src/__internal__/utils/rect.ts:3]: This class implementation looks trivial, we have added tests in `tl-utils` in phasor. So I suggest not to do that in this PR.

**Review by doodlewind — APPROVED** [review]

**Pull request merged**